### ISSUE TITLE
categorical_crossentropy with label smoothing support on tf.bfloat

### ISSUE THREAD

**Issue by tonychenxyz** [tensorflow/tensorflow#41095]
It seems that when enabling label smoothing, categorical_crossentropy only supports inputs in float32, so not useful when using tf.bfloat on TPU. I changed it to the following to make it work:
```
from tensorflow.python.framework import ops
from tensorflow.python.ops import math_ops
from tensorflow.python.framework import smart_cond
from tensorflow.python.keras import backend as K
from tensorflow.python.ops import array_ops

def categorical_crossentropy(y_true,
                             y_pred,
                             from_logits=False,
                             label_smoothing=0):

  y_pred = ops.convert_to_tensor_v2(y_pred)
  y_true = math_ops.cast(y_true, y_pred.dtype)
  label_smoothing = ops.convert_to_tensor_v2(label_smoothing, dtype=tf.bfloat16)

  def _smooth_labels():
    num_classes = math_ops.cast(array_ops.shape(y_true)[1], y_pred.dtype)
    return y_true * (1.0 - label_smoothing) + (label_smoothing / num_classes)

  y_true = smart_cond.smart_cond(label_smoothing,
                                 _smooth_labels, lambda: y_true)
  return K.categorical_crossentropy(y_true, y_pred, from_logits=from_logits)
```
